### PR TITLE
[feat] new success and verification pages with full functionality EXC…

### PR DESCRIPTION
**Pull request by RohinJ444**
…EPT automatic rerouting from verification to success is not yet functional

[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Old auth flow looked like this: sign-up --> sign-in --> onboarding 
- New auth flow looks like this: sign-up --> verification --> success --> onboarding

The verification page notifies users that their email needs to be verified, and provides them with options to use another account and to resend the verification link to their email. It displays pertinent error messages if any issues with resending the email. The success page confirms successful verification of user's email.

There remains one issue, which is that verification does not automatically redirect to success once the user's email is verified. Discussed this with Celine during worksession—the reason for this is that the auth session is still null before log in, but log in is not possible without successful verification. This makes retrieving user data difficult. Going to OH to debug/explore this further. However, the front-end styling and resend link functionality on the verification page.

Additionally, I slightly refactored styling, taking styles for these two new success and verification pages and placing them into a global styles/styles.ts file. The styled components are compatible with Melissa's onboarding pages, meaning that she should be able to import from global styles/styles.ts instead of the onboarding styles without issues. The sign-up and sign-in pages still use the styles in auth-styles.ts, so this can be refactored too as needed.

Finally, I fixed a bug with the auth context by creating a new function ensureLoggedOutForNewUser that ensures a user who is already signed-in is logged out when a new sign-up or sign-in is initiated. I am not preventing signed-in users from signing in with a new account or signing up for a new account, but rather, when signed in users do either of the following things, they will automatically be signed out and then signed in to the "newer" account.

### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

Screenshot 2024-12-05 at 1 15 22 PM

Screenshot 2024-12-05 at 1 16 46 PM

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Key front-end changes are in success and verification folders under app/auth, as well as the global styles/styles.ts file.

On the back-end side, I made the ensureLoggedOutForNewUser function in auth.ts under the api/supabase/queries folder.
I also made some small changes to back-end in api/supabase/queries/auth.ts and utils/AuthContextProvider.tsx (basically just storing the user's email address in localStorage during sign-up and retrieving it on the verification page so that the website can display the email address while the auth session is null before verification).

I would just test sign-up flow with the auth dashboard open https://supabase.com/dashboard/project/paifoskjfzbjbiehraqj/auth/users

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

As I mentioned, verification does not automatically redirect to success once the user's email is verified. Will update after OH.

Potentially might wanna restyle the error message on verification when there is no email address from signup. 

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii

### CODE REVIEW

File: app/(auth)/verification/page.tsx [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
 import { useEffect, useState } from 'react';
 import { useRouter } from 'next/navigation';
-import supabase from '@/api/supabase/createClient';
 import {
   getTempEmail,
   resendVerificationEmail,
 } from '@/api/supabase/queries/auth';
 import Bud from '@/public/images/bud.svg';
 import EmailIcon from '@/public/images/email.svg';
+// import supabase from '@/api/supabase/createClient';
```
Comment (celinechoiii) [inline app/(auth)/verification/page.tsx]: delete the commented line

File: app/(auth)/verification/page.tsx [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+import { useSession } from '@/utils/AuthProvider';
 import {
```
Comment (celinechoiii) [inline app/(auth)/verification/page.tsx:12]: replace `../../../styles/styles` to `@/styles/styles`

File: app/(auth)/verification/page.tsx
Comment (celinechoiii) [inline app/(auth)/verification/page.tsx:1]: instead of using `<text>` tags, import this line on the top of the file `import { P } from '@/styles/text;` and then use the `<P>` tag. you can adjust the font weight by passing in weight as a prop: `<P $fontWeight={300}>`. also, since styling such as `EmailContainer`, `EmailText` etc. are not being used in any other files, create a separate `styles.ts` file under `verification` since `styles/styles.ts` file is only for GLOBALLY used styled components.

Comment (RohinJ444) [reply app/(auth)/verification/page.tsx:1]: fixed in latest commit!

File: app/(auth)/verification/page.tsx (hunk 2 above)
Comment (RohinJ444) [reply app/(auth)/verification/page.tsx:12]: fixed in latest commit!

File: app/(auth)/verification/page.tsx (hunk 1 above)
Comment (RohinJ444) [reply app/(auth)/verification/page.tsx]: fixed in latest commit!
